rust/processor/postgres-init/init-db.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+
+psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" <<-EOSQL
+    DO
+    \$\$
+    BEGIN
+        IF NOT EXISTS (SELECT 1 FROM pg_roles WHERE rolname = '$POSTGRES_USER') THEN
+            CREATE ROLE $POSTGRES_USER WITH LOGIN PASSWORD '$POSTGRES_PASSWORD';
+        ELSE
+            ALTER ROLE $POSTGRES_USER WITH LOGIN PASSWORD '$POSTGRES_PASSWORD';
+        END IF;
+    END
+    \$\$;
+
+    ALTER ROLE $POSTGRES_USER CREATEDB;
+    GRANT ALL PRIVILEGES ON DATABASE $POSTGRES_DB TO $POSTGRES_USER;
+EOSQL
```

Comment:
Is there a reason this isn't a migration?

Also noting that for cloud compute this will probably error out, since operations like privileges sometimes have to be managed in cloud configs

Reply:
I think it should work, I basically took it from the official postgres docker image docs (link below)

The container runs the scripts in `postgrest-init` on the first mount of the db volume, and we can't run migrations without a user, so it has to be in here

ctrl+f on `docker-entrypoint-initdb.d/init-user-db.sh` here: https://hub.docker.com/_/postgres/